SConstruct

```diff
@@ -190,16 +200,50 @@ if unknown:
     for item in unknown.items():
         print("    " + item[0] + "=" + item[1])
 
+# Compatibility with old bits argument on Windows/Linux.
+# Not needed on macOS or web. Android and iOS don't use env["arch"].
+if env["platform"] in ["windows", "linux"]:
+    if env["bits"] == "64":
+        env["arch"] = "x86_64"
+    elif env["bits"] == "32":
+        env["arch"] = "x86_32"
+
+# Process CPU architecture argument.
+if env["arch"] == "auto":
+    # Use env["arch"] on all platforms except Android and iOS.
+    # For macOS, default arch to macos_arch, which defaults to universal (compat with old argument).
+    # For the web (javascript), default arch to wasm32.
+    # For Windows and Linux, default arch to the host architecture.
+    if env["platform"] == "osx":
+        env["arch"] = env["macos_arch"]
+    elif env["platform"] == "javascript":
+        env["arch"] = "wasm32"
+    else:
+        host_machine = platform.machine().lower()
+        if host_machine in architecture_array:
+            env["arch"] = host_machine
+        elif host_machine in architecture_aliases.keys():
+            env["arch"] = architecture_aliases[host_machine]
+        elif "86" in host_machine:
+            # Catches x86, i386, i486, i586, i686, etc.
+            env["arch"] = "x86_32"
+        else:
+            print("Unsupported CPU architecture: " + host_machine)
+            Exit()
+
+env_arch = env["arch"]
+
 # This makes sure to keep the session environment variables on Windows.
 # This way, you can run SCons in a Visual Studio 2017 prompt and it will find
 # all the required tools
 if host_platform == "windows" and env["platform"] != "android":
-    if env["bits"] == "64":
+    if env["arch"] == "x86_64":
         env = Environment(TARGET_ARCH="amd64")
-    elif env["bits"] == "32":
+    elif env["arch"] == "x86_32":
         env = Environment(TARGET_ARCH="x86")
 
     opts.Update(env)
+    env["arch"] = env_arch
```

Comment:
That kind of sucks. I think instead of `auto` we should implement the auto logic as `detect_arch()` and use that as the default value. So the default value automatically adjusts based on the detection and we don't need to override `env["arch"]` (so this `env_arch` workaround can be dropped).

Reply:
Done, it's now fully removed in favor of the default value being calculated with `detect_arch()`.